### PR TITLE
Vendor: github.com/Azure/azure-docker-extension

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
Vendor-only change.

Incorporate latest changes (PRs 96, 97, 98, 99) from azure-docker-extension repo
where we grab extension handler authoring utility pkgs for Go.

### CODE REVIEW

**Comment by ahmetalpbalkan** [comment]
Build fails but the next PR will fix it...

File: vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlerenv.go [hunk 1]
```diff
@@ -3,8 +3,18 @@ package vmextension
 import (
 	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
 )
 
+// HandlerEnvFileName is the file name of the Handler Environment as placed by the
```
Comment (boumenot) [inline vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlerenv.go]: placed => created ?

File: vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlersettings.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	"io/ioutil"
 	"path/filepath"
 
-	"github.com/Azure/azure-docker-extension/pkg/executil"
+	"os/exec"
```
Comment (boumenot) [inline vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlersettings.go]: extraneous new line

File: vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlerenv.go (hunk 1 above)
Comment (ahmetalpbalkan) [inline vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlerenv.go]: meh, both are the same in this case. also if we need to fix it, it'd be better to fix it upstream

**Comment by boumenot** [comment]
LGTM.

File: vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlersettings.go (hunk 2 above)
Comment (ahmetalpbalkan) [inline vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlersettings.go]: ah fair vscode keeps failing me on imports, but I'll keep it the same as it is harmless for now.

File: vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlerenv.go (hunk 1 above)
Comment (boumenot) [inline vendor/github.com/Azure/azure-docker-extension/pkg/vmextension/handlerenv.go]: You're right.  I forgot this was a vendor'ed.

**Pull request merged**